### PR TITLE
fix: fixed diff test 11885

### LINKED ISSUE #11885: expected 'INVALID_ALLOWANCE_SPENDER_ID', got 'INVALID_ACCOUNT_ID' (round 162926130)
### Description

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708343833.983966003)

Transaction Body:
```
transactionID {
  transactionValidStart {
    seconds: 1708343811
    nanos: 75933899
  }
  accountID {
    accountNum: 2189934
  }
}
nodeAccountID {
  accountNum: 24
}
transactionFee: 10000000000
transactionValidDuration {
  seconds: 120
}
cryptoApproveAllowance {
  tokenAllowances {
    tokenId {
      tokenNum: 4381245
    }
    owner {
      accountNum: 2189934
    }
    amount: 1000000
  }
}
```

### Steps to reproduce

Re-run events from round 162926130

### Additional context

Possibly related to #11876 

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn

### Operating system

macOS

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Fixed expected 'INVALID_ALLOWANCE_SPENDER_ID', got 'INVALID_ACCOUNT_ID' error on diff test 11885

**Related issue(s)**:

Fixes #11885

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java [hunk 1]
```diff
@@ -113,6 +115,7 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
 
         for (final var allowance : tokenAllowances) {
             validateNullableAccountID(allowance.owner());
+            validateFalsePreCheck(allowance.amount() > 0 && allowance.spender() == null, INVALID_ALLOWANCE_SPENDER_ID);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java:118]: We should probably refactor to pass the response code `INVALID_ALLOWANCE_SPENDER_ID)` into `validateAccountID` on line 119. It is already doing the same check imo. This line is not needed.

Comment (iwsimon) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java:118]: allowance.amount() was not checked in `validateAccountID`. Should I also include this check on line 112 too as I saw similar check in mono [here](https://github.com/hashgraph/hedera-services/blob/8aeb2d68d61574e74f77b728211a8001ab38f8cb/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java#L113).

Comment (iwsimon) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java]: Updated the change.

**Review by kimbor — APPROVED** [review]
lgtm!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @iwsimon

**Pull request merged**